### ISSUE TITLE
Accuracy of Eyes compared to Dlib

### ISSUE THREAD

**Issue by jtressle** [1adrianb/face-alignment#45]
Hi, 

Great piece of software.  It definitely gets better poses than other solution.

One thing I noticed over 10 images is that the eye point accuracy isn't as accurate as dlib's feature extraction.  Is this a by-product of the training?  And is this also seen with the Lua version?  

Here's an example of one of the worst alignments.

Thanks,

**Face-Alignment**
![face_alignment](https://user-images.githubusercontent.com/13778165/34976480-b1e904ea-fa4b-11e7-9019-4b83f6e55363.png)

**Dlib**
![dlib](https://user-images.githubusercontent.com/13778165/34976481-b2052f6c-fa4b-11e7-9aaa-e8e2acc30150.png)

**Comment by 1adrianb** [comment]
Hi @jtressle 
If you can send me an input image other email (of course if one of  the images can be shared) I can definetly have a look a check where the issue arise from.

It is likely however that the eyes may not be as accurate, since the dataset used for training (300W-LP) often didn't account for closed or open eyes nor did it mached them well in certain cituations (the dataset was  synthetically warped in larger poses, starting from a frontal or close to frontal one).

**Comment by jtressle** [comment]
Hi Adrian,

Thanks for the response.  Here's the image in question.  It has the worst
discrepancy.

That makes sense about the training.  I did find the 'Helen' dataset to be
sufficient for some of my training.  However, I didn't warp the images.

Thanks,
Jan-Michael

**Comment by 1adrianb** [comment]
Hi @jtressle 
Unfortunatelly it seems that github removed the attachment. Could you please resend it to adrian.bulat@nottingham.ac.uk?

Helen unfortunatelly contains only frontal or near-frontal poses and  was unsuitable for large poses. I mean the warped image are also not ideal, but the network manages to ignore most of the noise introduced by the warping process so they are good starting point. 

Thanks,
Adrian

**Comment by HavenFeng** [comment]
Hi @1adrianb , 
we also find this problem, the performance on eye region is worse than the lua version and even Dlib. BTW, although the new pre-trained model is way better than the previous one, landmark keeps shaking on a sequence(which doesn't include obvious head movement). maybe you have some idea around this.

Thanks,
Haven

**Comment by 1adrianb** [comment]
Hi @HavenFeng ,
This is caused by a) the 64x64 ouput resolution and b) the annotations from the training set (300W-LP). On the lua side , one of the models was finetuned on 300W and as such should have a higher accuracy for the eyes.

Regarding the shaking problem you can either try to apply some temporal smothing or intialise the bouding box from the previous succesful fitting. 

Thanks,
Adrian
